cmd/spire-server/cli/agent/list.go

```diff
@@ -1,104 +1,86 @@
-	if c.RegistrationUDSPath == "" {
-		return errors.New("a socket path for registration api is required")
-	}
-	return nil
+// NewListCommand creates a new "list" subcommand for "agent" command.
+func NewListCommand() cli.Command {
+	return newListCommand(common_cli.DefaultEnv, util.NewClients)
 }
 
-//ListCLI command for listing attested nodes
-type ListCLI struct {
-	registrationClient registration.RegistrationClient
-	nodeList           []*common.AttestedNode
+func newListCommand(env *common_cli.Env, clientsMaker util.ClientsMaker) cli.Command {
+	return util.AdaptCommand(env, clientsMaker, new(listCommand))
 }
 
-func (ListCLI) Synopsis() string {
-	return "Lists attested agents and their SPIFFE IDs"
+func (*listCommand) Name() string {
+	return "agent list"
 }
 
-func (c ListCLI) Help() string {
-	_, err := c.parseConfig([]string{"-h"})
-	return err.Error()
+func (listCommand) Synopsis() string {
+	return "Lists attested agents and their SPIFFE IDs"
 }
 
-//Run will lists attested agents
-func (c *ListCLI) Run(args []string) int {
-	ctx := context.Background()
-
-	config, err := c.parseConfig(args)
+//Run lists attested agents
+func (c *listCommand) Run(ctx context.Context, env *common_cli.Env, clients *util.Clients) error {
+	listResponse, err := clients.AgentClient.ListAgents(ctx, &agent.ListAgentsRequest{})
 	if err != nil {
-		fmt.Fprintln(os.Stderr, err)
-		return 1
+		return err
 	}
-
-	if err = config.Validate(); err != nil {
-		fmt.Fprintln(os.Stderr, err)
-		return 1
+	c.agents = listResponse.Agents
+	if len(c.agents) == 0 {
+		return env.Printf("No attested agents found\n")
 	}
 
-	if c.registrationClient == nil {
-		c.registrationClient, err = util.NewRegistrationClient(config.RegistrationUDSPath)
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "Error establishing connection to the Registration API: %v \n", err)
-			return 1
-		}
-	}
+	msg := fmt.Sprintf("Found %d attested ", len(c.agents))
+	msg = util.Pluralizer(msg, "agent", "agents", len(c.agents))
+	env.Printf(msg + ":\n\n")
 
-	listResponse, err := c.registrationClient.ListAgents(ctx, &registration.ListAgentsRequest{})
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Error listing attested agents: %v \n", err)
-		return 1
-	}
-	c.nodeList = listResponse.Nodes
-	c.printAttestedNodes()
-	return 0
+	return printAgents(c.agents, env)
 }
 
-func (ListCLI) parseConfig(args []string) (*ListConfig, error) {
-	f := flag.NewFlagSet("agent list", flag.ContinueOnError)
-	c := &ListConfig{}
-
-	f.StringVar(&c.RegistrationUDSPath, "registrationUDSPath", util.DefaultSocketPath, "Registration API UDS path")
-
-	return c, f.Parse(args)
+func (c *listCommand) AppendFlags(fs *flag.FlagSet) {
 }
 
-func (c ListCLI) printAttestedNodes() {
-	msg := fmt.Sprintf("Found %d attested ", len(c.nodeList))
-	msg = util.Pluralizer(msg, "agent", "agents", len(c.nodeList))
-	fmt.Printf(msg + ":\n\n")
+func printAgents(agents []*types.Agent, env *common_cli.Env) error {
```

Comment:
nit: This might be more ergonomic to use if it were changed like so:
```suggestion
func printAgents(env *common_cli.Env, agents ...*types.Agent) error {
```

It would be a little easier to print out a single agent in the `show` case this way